src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private static final int ADD_ID = 1;
+    private static final int REMOVE_ID = 0;
+    private T[] storage;
+    private int size;
+    private int currentCapacity;
+
+    {
+        storage = (T[]) new Object[INITIAL_SIZE];
+        currentCapacity = storage.length;
+    }
```

Comment:
You can wrap these two lines into
public ArrayList() {
} for example. To make this part of code more readable.